docs/examples/lipstick-pig.md

```diff
@@ -0,0 +1,315 @@
+This example is heavily inspired by the [lipstick on a pig paper](https://arxiv.org/pdf/1903.03862.pdf)
+by Hila Gonen and Yoav Goldberg.
+
+## Meaning
+
+In word embedding space you might wonder if a direction in embedding space might represents meaning.
+
+- $v_{man}$ represents the word embedding for "man" .
+- $v_{woman}$ represents the word embedding for "woman" .
+- You could argue that the axis $v_{man} - v_{woman}$ might represent "gender".
+
+One side of the "gender" axis would represent the male gender while the other end represents the female gender. This spectrum might allow you to guess if words are more "male" or more "female". It can also be used as a measurement for bias in language.
+
+## Similarity
+
+There are some axes that we might come up with that should be similar to this gender axis.
+
+- $v_{he} - v_{she}$
+- $v_{king} - v_{queen}$
+- $v_{brother} - v_{sister}$
+
+It would be highly unfortunate though if the following pairs of words would display similarity.
+
+- $v_{nurse} - v_{physician}$
+- $v_{nurse} - v_{surgeon}$
+- $v_{nurse} - v_{doctor}$
+
+Being a woman should not imply that you are a nurse just like that being a man should not imply that you are a surgeon. It'd be a shame if we were using embeddings where such stereotypical bias present. Unfortunately, it's likely in the embeddings. Historically woman have not gotten the same opportunities as men. This is bound to be reflected on websites like Wikipedia which are a common source of data to train word embeddings. So let's make a similarity chart to confirm if this is the case.
+
+<details>
+  <summary><b>Word Pairs.</b></summary>
+```python
+stereotype_pairs = [
+    ('sewing', 'carpentry'),
+    ('nurse', 'physician'),
+    ('nurse', 'surgeon'),
+    ('nurse', 'doctor'),
+]
+
+appropriate_pairs = [
+    ('woman', 'man'),
+    ('she', 'he'),
+    ('her', 'him'),
+    ('girl', 'boy')
+]
+
+random_pairs = [
+    ('dog', 'firehydrant'),
+    ('carpet', 'leg'),
+    ('hot', 'cold'),
+]
+
+all_pairs = [stereotype_pairs, appropriate_pairs, random_pairs]
+```
+</details>
+
+```python
+from whatlies import Embedding, EmbeddingSet
+from whatlies.language import FasttextLanguage
+
+lang_ft = FasttextLanguage("cc.en.300.bin")
+
+flatten = lambda l: [item for sublist in l for item in sublist]
+
+def calc_axis(pair_list, language_model):
+    return [language_model[t1] - language_model[t2] for (t1, t2) in pair_list]
+
+def make_correlation_plot(pairs, language_model, metric="cosine"):
+    axes = [calc_axis(p, language_model) for p in pairs]
+    emb_pairs = EmbeddingSet(*flatten(axes))
+    emb_pairs.plot_distance(metric=metric)
+
+make_correlation_plot(pairs=all_pairs, language_model=lang_ft)
+```
+
+This code generates a similarity chart for fasttext embeddings, shown below.
+
+![](imgs/bias.png)
+
+Notice, that we indeed see correlation. The "gender" direction seems to correlate with the "doctor-nurse" direction as well. This does not bode well.
```

Comment:
Is the correlation really that strong? Because, to me, doesn't look like so! Maybe I am looking at the wrong place! Specifically, are you referring to "she-he" vs. "nurse-physician" (or vs. "nurse-surgeon")? Because except those two, I don't see a strong correlation in terms of gender bias?!!

Reply:
I've replaced it with: 

> It's not a high correlation, but we'd prefer if it simply were zero.

Reply:
Fine! But I hope this one is not confusing either for the reader (i.e. zero for correlation, not zero distance).